plugins/modules/mso_schema_template_contract_filter.py

```diff
@@ -240,20 +240,33 @@ def main():
 
     if contract in contracts:
         contract_idx = contracts.index(contract)
+        contract_obj = schema_obj.get('templates')[template_idx]['contracts'][contract_idx]
 
         filters = [f.get('filterRef') for f in schema_obj.get('templates')[template_idx]['contracts'][contract_idx][filter_key]]
         filter_ref = mso.filter_ref(schema_id=filter_schema_id, template=filter_template, filter=filter_name)
         if filter_ref in filters:
             filter_idx = filters.index(filter_ref)
             filter_path = '/templates/{0}/contracts/{1}/{2}/{3}'.format(template, contract, filter_key, filter_name)
-            mso.existing = schema_obj.get('templates')[template_idx]['contracts'][contract_idx][filter_key][filter_idx]
+            mso.existing = contract_obj.get(filter_key)[filter_idx]
+            mso.existing['filterRef'] = mso.dict_from_ref(mso.existing.get('filterRef'))
+            mso.existing['displayName'] = contract_obj.get('displayName')
+            mso.existing['filterType'] = filter_type
```

Comment:
Aren't we overriding a filterType with our user-input in the query part?

Reply:
yes, we do get a filtertype from input. because we can only get contract_filter_type from api.
the filter type is parameter only used to decide the filter path and we can't get it from api.
do you think it's better to remove this from existing or just keep it ?

Reply:
We can just keep it.